lib/manager/gomod/update.ts

```diff
@@ -94,7 +96,12 @@ export function updateDependency(
       }
     }
     if (lineToChange.endsWith('+incompatible')) {
-      newLine += '+incompatible';
+      let toAdd = '+incompatible';
+
+      if (upgrade.updateType === 'major' && upgrade.newMajor >= 2) {
+        toAdd = '';
+      }
+      newLine += toAdd;
```

Comment:
@mangas Any documentation for that?

Reply:
Same link has above. "incompatible" is used for when a package has not been migrated to go mod and or there is a version higher than 2

Reply:
In this case if we;'re doing a major update, if i understand this correctly, the tag is no longer needed

Reply:
https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-v2-package-that-has-not-opted-into-modules-what-does-incompatible-mean